src/formats.rs

```diff
@@ -12,20 +12,35 @@ pub enum InputFormat {
 }
 
 impl InputFormat {
-    pub fn load_programs_and_tasks(&self, path: &Path) -> Result<(Vec<String>, Vec<String>, usize), String> {
+    pub fn load(&self, path: &Path, split_train_test: bool)
+    -> Result<(Vec<String>, Option<Vec<String>>, Vec<String>, usize), String> {
         match self {
             &InputFormat::Dreamcoder => {
                 // read dreamcoder format
                 let json: serde_json::Value = from_reader(File::open(path).expect("file not found")).expect("json deserializing error");
-                let frontiers = json["frontiers"].as_array().unwrap_or_else(||panic!("json parse error, are you sure you wanted format {:?}?", self));
                 let dsl: Vec<String> = json["DSL"]["productions"].as_array().unwrap().iter().map(|prod|prod["expression"].as_str().unwrap().to_string()).collect();
                 let inv_dc_strs: Vec<(String,String)> = dsl.iter().rev() // rev() since the first dsl function is the newest
                     .filter(|s| s.starts_with("#")).cloned().enumerate()
                     .map(|(i,dc_str)| (format!("fn_{}",i),dc_str)).collect();
                 let num_prior_inventions: usize = inv_dc_strs.len();
-                let mut programs: Vec<String> = Vec::default();
+
+                // in the DreamCoder format, a train/test split will be provided through the 'topK_frontiers' field,
+                // if it is provided
+                let mut train_programs: Vec<String> = Vec::default();
+                let mut test_programs: Option<Vec<String>> = None;
                 let mut tasks: Vec<String> = Vec::default();
-                for (i,frontier) in frontiers.into_iter().enumerate() {
+                let train_frontiers;
+                let test_frontiers;
+                if split_train_test {
+                    train_frontiers = json["topK_frontiers"].as_array().unwrap_or_else(||panic!("json parse error, are you sure you wanted format {:?}?", self));
```

Comment:
OK to hard code this? Could also take the fields to use for the training and test as cmd-line arg strings, but that's a lot of (error-prone?) args to pass...